godot-core/src/builtin/array.rs

```diff
@@ -261,65 +465,33 @@ impl Array {
         }
     }
 
-    /// Returns the minimum value contained in the array if all elements are of comparable types.
-    /// If the elements can't be compared or the array is empty, `None` is returned.
-    pub fn min(&self) -> Option<Variant> {
-        let min = self.as_inner().min();
-        (!min.is_nil()).then_some(min)
-    }
-
-    /// Returns the maximum value contained in the array if all elements are of comparable types.
-    /// If the elements can't be compared or the array is empty, `None` is returned.
-    pub fn max(&self) -> Option<Variant> {
-        let max = self.as_inner().max();
-        (!max.is_nil()).then_some(max)
-    }
-
-    /// Returns a random element from the array, or `None` if it is empty.
-    pub fn pick_random(&self) -> Option<Variant> {
-        (!self.is_empty()).then(|| self.as_inner().pick_random())
-    }
-
-    /// Sets the value at the specified index as a `Variant`. To convert a specific type (which
-    /// implements `ToVariant`) to a variant, call [`ToVariant::to_variant`] on it.
+    /// Sets the value at the specified index.
     ///
     /// # Panics
     ///
     /// If `index` is out of bounds.
-    pub fn set(&mut self, index: usize, value: Variant) {
+    pub fn set(&mut self, index: usize, value: T) {
         let ptr_mut = self.ptr_mut(index);
         // SAFETY: `ptr_mut` just checked that the index is not out of bounds.
         unsafe {
-            *ptr_mut = value;
+            *ptr_mut = value.to_variant();
         }
     }
 
     /// Appends an element to the end of the array. Equivalent of `append` and `push_back` in
     /// GDScript.
-    pub fn push(&mut self, value: Variant) {
-        self.as_inner().push_back(value);
+    ///
+    /// This takes `value` by reference because it needs to be converted to a `Variant` anyway.
+    pub fn push(&mut self, value: T) {
+        self.as_inner().push_back(value.to_variant());
     }
```

Comment:
Outdated doc? `value` is passed by value.

See main comment regarding design.

Reply:
Done.